src/main/java/com/autotune/database/dao/ExperimentDAOImpl.java

```diff
@@ -75,6 +78,43 @@ public ValidationOutputData addExperimentToDB(KruizeExperimentEntry kruizeExperi
         return validationOutputData;
     }
 
+    @Override
+    public void deletePartitions(int daysCount) {
+        LOGGER.info("Threshold is set to {}", daysCount);
+        // Calculate the date 'daysCount' days ago
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, -daysCount);
+        Date cutoffDate = calendar.getTime();
+
+        try (Session session = KruizeHibernateUtil.getSessionFactory().openSession()) {
+            List<String> tablenames = session.createNativeQuery(SELECT_ALL_KRUIZE_TABLES).getResultList();
+            for (String tableName : tablenames) {
+                String datePart = null;
+                if (tableName.startsWith("kruize_results_")) {
+                    datePart = tableName.substring("kruize_results_".length());
+                } else if (tableName.startsWith("kruize_recommendations_")) {
+                    datePart = tableName.substring("kruize_recommendations_".length());
```

Comment:
Suggest you use constants for all of these strings

Reply:
done